tests/flytekit/unit/core/test_gate.py

```diff
@@ -290,3 +291,36 @@ def cond_wf(a: int) -> float:
         x = cond_wf(a=3)
         assert x == 6
         assert stdin.read() == ""
+
+
+def test_basic_sleefdsap():
+    @task
+    def t1(a: int) -> int:
+        return a + 5
+
+    @task
+    def t2(a: int) -> int:
+        return a + 6
+
+    @workflow
+    def wf(a: int) -> typing.Tuple[int, int, int]:
+        zzz = sleep(timedelta(seconds=10))
+        x = t1(a=a)
+        s1 = wait_for_input("my-signal-name", timeout=timedelta(hours=1), expected_type=bool)
```

Comment:
qq: should the name of the gate node be unique?

Reply:
so this is actually an interesting question. signals ids are unique within a workflow execution. so if you had this same statement twice it would just wait twice for a single signal with the given name. nothing would break unless the expected types are different. presumably you could have two subworkflows that both wait on the same signal by using the same name. at least that's how it works in the backend. not sure if we want to explicitely restrict this from flytekit side?

Reply:
got it, thanks for the explain

Reply:
yes.